### ISSUE TITLE
[Debug] Any way to inspect the compiled program?

### ISSUE THREAD

**Issue by aditvenk** [ml-explore/mlx#2683]
I have a fairly decomposed MLX program (for instance, softmax is fully decomposed into primitive ops) which I am able to run using default mode & mlx.compile. With compile I see a slight speedup, but not a lot. 

I am wondering if there is any way to inspect the actual compiled artifacts to see what stuff is getting fused and what isn't? 

For reference, I am wondering if there is some analogue to PyTorch compiler's `TORCH_LOGS="output_code"` which dumps the output code (e.g., Triton,) emitted by the compiler.

**Comment by aditvenk** [comment]
Update: from following the steps in https://ml-explore.github.io/mlx/build/html/dev/metal_debugger.html , I could capture the Metal trace which shows the Metal shader source for the compiled program.

Pl feel free to close this issue, unless there is a more convenient way to examine compiled sources, or maybe the fused IR (perhaps before codegen to Metal sources).

**Comment by zcbenz** [comment]
I'm closing this issue as the user had already found a solution.
